platform/src/main/python/dlpx/virtualization/platform/_linked.py

```diff
@@ -112,6 +112,35 @@ def source_size_decorator(source_size_impl):
 
         return source_size_decorator
 
+    @staticmethod
+    def _from_protobuf_remote_mount(remote_mount):
+        return Mount(remote_environment=RemoteEnvironment.from_proto(
+            remote_mount.remote_environment),
+            mount_path=remote_mount.mount_path,
+            shared_path=remote_mount.shared_path)
+
+    @staticmethod
+    def _get_mounts_from_request(request):
+        staged_mount = request.staged_source.staged_mount
+        mounts = request.staged_source.mounts
+        if LinkedOperations._check_staged_mount(staged_mount) \
+                and mounts and len(mounts) > 0:
+            raise PluginRuntimeError(
+                'Either staged_mount or mounts can be present for staging source. '
+                'Found both staged_mount and mounts.')
+
+        if mounts and len(mounts) > 0:
+            return None, [
+                LinkedOperations._from_protobuf_remote_mount(m) for m in mounts]
+        else:
+            return LinkedOperations._from_protobuf_remote_mount(staged_mount), None
+
+    @staticmethod
+    def _check_staged_mount(staged_mount):
```

Comment:
Should we say this method `is_valid_mounts_passed_in_the_request` and include the `and mounts and len(mounts) > 0:` check also in the same function?

This might make it more readable and centralized logic.

Reply:
As discussed, from mount specification we do pass only mounts when plugin provides more than 1 mounts. But as it is part of request, we just check if `mounts` or `staged_mount` is present in the request sent by Delphix and set it accordingly to staged_source.

Reply:
Removed the method all together as there was no duplication of code. Added it as part of _get_mounts_from_request() method itself.